pkg/util/discovery/server_resourcer.go

```diff
@@ -64,6 +65,11 @@ func GetResources(discoveryClient ServerResourcer) ([]*metav1.APIResourceList, s
 			for gv, subErr := range discoErr.Groups {
 				if apierrors.IsNotFound(subErr) {
 					klog.Warningf("Failed to discover APIGroups %s due to not found, moving on: %v", gv, subErr)
+				} else if strings.Contains(subErr.Error(), "received empty response") {
+					// Allow "empty response" as common responses for metrics
+					// Upstream workaround https://github.com/kubernetes/kubernetes/pull/115978
```

Comment:
Can you make an upstream PR to make `emptyResponseError` public, so we can catch it more explicitly in the future, then add a TODO here?

Reply:
Made https://github.com/kubernetes/kubernetes/pull/126137, also created http://b/353596666 as TODO.